active-rfcs/0000-style-unification.md

```diff
@@ -0,0 +1,91 @@
+- No longer supported to use with `Astro.resolve()`! (see other options below for how to reference a style that lives inside of `src/`)
+- Not touched by Astro compiler, runtime, or Vite. Both the `<link>` tag and the referenced CSS file will be left as-is in the final output. 
+- Must point to a file in `public/` or an external CSS URL. This will not be bundled/optimized.
+- Must be nested inside of a `<head>` element.
+
+Note: See the [local directive RFC](https://github.com/withastro/rfcs/blob/build-performance-rfc/active-rfcs/0000-build-performance.md#local-directive) for another alternative on referencing a `src/` file that would be processed/built/bundled.
+
+
+#### 2. ❌ `<link rel="stylesheet" href={xUrl}> // import xUrl from './X.css?url';`
+
+- Note: Vite currently doesn't support a CSS `?url` import. @matthewp fixed a bug where this now returns the correct URL for `xUrl`, but it still won't bundle this CSS file as you would expect. This means that the resolved URL might not actually exist in the final build, if that file had been bundled. 
+- It would be great to support this once Vite fixes their issues, but right now this is considered a Vite limitation. @matthewp to confirm for the final RFC.
+- reproduction: https://stackblitz.com/edit/vite-1jaulg?file=dist%2Fassets%2Findex.dd7a6b16.js
+
+<!-- 
+- Still supported!
+- Vite will see the ESM import, and make sure that this asset exists in your final build.
+- Note: There is currently a Vite bug in this support for `import './foo.css?url'`, see: https://stackblitz.com/edit/vite-1jaulg?file=dist%2Fassets%2Findex.dd7a6b16.js
+-->
+
+#### 3. ✅ `<style>@import './X.css';</style>` or  `<style global>@import './X.css';</style>`
+
+- **Usecase:** Importing a CSS file that you want optimized with the rest of your site.
+- Imported CSS is bundled with the rest of the Astro component-level CSS on the page.
+- (for scoped `<style>`) - Imported CSS is not scoped. Only the CSS defined inline inside the `<style>` element is scoped.
+- Bundled with the rest of your final CSS. Vite will make sure that duplicate imports are resolved across the final page CSS.
+- Resolved by Vite, which means that `@import './X.css';` will inherit from all Vite support, including aliases and npm package `@import` support.
+
+#### 4. ✅ `import './X.css'; // in Astro component frontmatter`
+
+- Supported as a Vite side-effect, but we would not really document this and instead recommend 3. in our documentation and examples.
+- Same build/bundle behavior as if you had done 3.
+- In a previous draft, the proposal was to completely disable this. However, we wouldn't be able to disable you importing a JS file that then did it's own CSS file import, so we decided to just leave the support in for those who ended up using it.
+
+#### 5. ✅ `import './X.css'; // in React component`
+
+- **Usecase:** Supported as the only way to reference a CSS file inside of a React component.
```

Comment:
Nit, this also works in Preact, Solid, Lit, or any 3rd party JavaScript-based component frameworks.